### PR TITLE
Miscellaneous tasks for 0.42.0 release - New compiler information

### PR DESCRIPTION
**Pull request by Sreekala-Gopakumar**
Included the new compiler information for Java 21 in the Migrating from Java 17 to Java 21 topic.

Signed-off-by: Sreekala Gopakumar <sreekala.gopakumar@ibm.com>

### CODE REVIEW

**Comment by Sreekala-Gopakumar** [comment]
Jenkins doc stage

**Comment by Sreekala-Gopakumar** [comment]
Jenkins doc stage

**Comment by Sreekala-Gopakumar** [comment]
@pshipton - I have added the new compiler information for Java 21 in the Migrating from Java 17 to Java 21 topic. Please check and confirm. Thanks!

**Comment by pshipton** [comment]
The Windows VS compiler changed as well from VS2019 to VS2022.

File: docs/migrating17to21.md [hunk 1]
```diff
@@ -48,3 +48,4 @@ The following table lists the new OpenJ9 features and notable changes with the O
 | New `-XX:[+\|-]ShowCarrierFrames` option added. You can use the `-XX:+ShowCarrierFrames` option to add the stack trace of the carrier thread in addition to the virtual thread stack trace to the `Throwable.getStackTrace()` method, if an exception occurs. | 0.41.0  |
 | New `-XX:ContinuationCache` option added. You can optimize the virtual thread performance by tuning the continuation tier 1 and 2 cache size with the `-XX:ContinuationCache` option | 0.41.0  |
 | Warnings are issued when the agents are loaded dynamically into a running VM after startup without specifying the `-XX:+EnableDynamicAgentLoading` option and the same agents were not loaded before. |  0.41.0    |
+| Linux&reg; builds for all platforms, except for the AArch64 64-bit, use gcc 11.2 instead of gcc 10.3. Linux AArch64 64-bit continues to use gcc 10.3. See the list of [build environments](openj9_support.md#build-environments). | 0.43.0 |
```
Comment (pshipton) [inline docs/migrating17to21.md:51]: It should be 0.42 not 0.43 for the change for jdk21.

**Comment by pshipton** [comment]
Pls squash the commits.

**Comment by Sreekala-Gopakumar** [comment]
Jenkins doc stage

**Comment by Sreekala-Gopakumar** [comment]
> Pls squash the commits.

Done, Peter.

**Review by pshipton — APPROVED** [review]

**Pull request merged**